### PR TITLE
Bug 1177110 - Fix take picture to use updated error types.

### PR DESCRIPTION
**Pull request by aosmond**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=dc49e1a6222e5d823e54de179708734d5cde3be6

File: apps/camera/js/lib/camera/camera.js [hunk 1]
```diff
@@ -765,7 +765,7 @@ Camera.prototype.takePicture = function(options) {
 
     // if taking a picture fails because there's
     // already a picture being taken we ignore it.
-    if (error === 'TakePictureAlreadyInProgress') {
+    if (error.name === 'NS_ERROR_IN_PROGRESS') {
```
Comment (justindarc) [inline apps/camera/js/lib/camera/camera.js]: Do we have any more of these types of comparisons for errors elsewhere?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4ccc972e2ce2ee5cc8c28ac49ead2e0ea976f820

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=69f6c04512970a48393f9073fbc3939873fdfbe8

**Pull request merged**